### ISSUE TITLE
Disabling required extension doesn't work from child config

### ISSUE THREAD

**Issue by aliaksandrb** [rubocop-hq/rubocop#7640]
Hi!

It looks like disabling an extension required in parent config doesn't work from the child config.

Assuming we have parent-child configs relationships as following:

Parent, which is shared among many other projects:
```yml
require:
  - rubocop-rspec
  - rubocop-rails
...
```

Child, in particular project which doesn't use rspec:
```yml
inherit_from: shared/.rubocop.yml

RSpec:
  Enabled: false
...
```

Runtime details:
```
ruby 2.5.1p57
rubocop (0.74.0)
rubocop-rails (2.2.1)
rubocop-rspec (1.35.0)
```

--------

## Expected behavior

Definition in the child config should take precedence over required directive.  So the expected behavior would be to suppress all violations related to disabled extension.

## Actual behavior

Running Rubocop in debug mode produces the following info:

```bash
$ rubocop -E -S -d --format simple

For /: configuration from /.rubocop.yml
Inheriting configuration from /example/.rubocop.yml
configuration from /usr/local/bundle/gems/rubocop-rspec-1.35.0/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rspec-1.35.0/config/default.yml
Default configuration from /usr/local/bundle/gems/rubocop-0.74.0/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rails-2.2.1/config/default.yml
configuration from /usr/local/bundle/gems/rubocop-rails-2.2.1/config/default.yml
For /example configuration from /example/.rubocop.yml
AllCops/Exclude configuration from /.rubocop.yml
Inheriting configuration from /example/.rubocop.yml
```

and some violations detected for `rspec` extension (reduced for clarity):
```
== example/spec/runner_spec.rb ==
C: 32:  7: Example has too many lines [6/5]. (http://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ExampleLength)
```

As a result it reduces the power of the configs inheritance as soon as you have an extension used by all but a few children.

## Important note

It does work as expected if disabling happens in the same parent file. For example:

Parent:
```yml
require:
  - rubocop-rspec
  - rubocop-rails

RSpec:
  Enabled: false
...
```

Child:
```yml
inherit_from: shared/.rubocop.yml
...
```
 
Works as expected (rspec related violations are being suppressed).

**Comment by buehmann** [comment]
@aliaksandrb Thanks for your report. Could you put together a complete example? I tried to reproduce your issue with the information given but failed. See here: https://github.com/buehmann/rubocop-issues/tree/7640

```
➜  rubocop-issues (7640) bundle exec rubocop
Inspecting 2 files
..

2 files inspected, no offenses detected
```
and when I enable the `RSpec` department in `.rubocop.yml`:
```
➜  rubocop-issues (7640) ✗ git diff
diff --git a/.rubocop.yml b/.rubocop.yml
index fd57358..4a111f4 100644
--- a/.rubocop.yml
+++ b/.rubocop.yml
@@ -1,4 +1,4 @@
 inherit_from: shared/.rubocop.yml

 RSpec:
-  Enabled: false
+  Enabled: true
➜  rubocop-issues (7640) ✗ bundle exec rubocop
Inspecting 2 files
.C

Offenses:

spec/runner_spec.rb:4:3: C: RSpec/ExampleLength: Example has too many lines [11/5].
  it 'foo' do ...
  ^^^^^^^^^^^

2 files inspected, 1 offense detected
```

**Comment by aliaksandrb** [comment]
Hi @buehmann, thanks for looking into it!

I've played around with your repository and found the difference with my setup, which had shown that the problem tends to be on my side.

In fact, my directory structure is more complicated and looks like rails one (meta gem with multiple gems within it). And between the root `/.rubocop.yml` and `/xxx/yyy/spec/runner_spec.rb` file there is `/xxx/.rubocop.yml` which has no
```yml
RSpec:
  Enabled: false
```

I was thinking that rubocop adopts config relative to the `$pwd` it being ran, but it looks like it checks the closest config relative to the every file being analyzed (wow, thats a lot of parsing if you have a mono-repo with many configs).

So my fault, works as expected 👌 I am closing this issue.

**Comment by buehmann** [comment]
Good to hear that. Yes, it is the closest config that is relevant for every file, but RuboCop makes sure to parse every config only once. That config is then reused for all files checked in the subtree (unless there is another config in a subdirectory, of course).
